cinema/views.py

```diff
@@ -56,3 +93,64 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    @staticmethod
+    def _param_to_date(query_string):
+        return datetime.datetime.strptime(query_string, "%Y-%m-%d")
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("movie", "cinema_hall")
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows")
+                        * F("cinema_hall__seats_in_row")
+                        - Count("tickets")
+                    )
+                )
+            )
+        if self.action == "retrieve":
+            queryset = (
+                queryset
+                .select_related("movie", "cinema_hall")
+                .prefetch_related("tickets")
+            )
+
+        date = self.request.query_params.get("date")
+        if date:
+            queryset = queryset.filter(
+                show_time__date=date
```

Comment:
this works if the format is exactly as expected and because Django tries to cover us, but the param type should be date here